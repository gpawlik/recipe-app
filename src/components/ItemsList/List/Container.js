import styled from 'styled-components';

const Container = styled.ul`
  margin: 0 auto;
  padding: 10px;
  list-style-type: none;
  max-width: 800px;
`;

export default Container;
