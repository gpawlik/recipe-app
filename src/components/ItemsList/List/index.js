import React from 'react';
import PropTypes from 'prop-types';
import shortid from 'shortid';

import Container from './Container';
import Item from 'components/ItemsList/Item';
import EmptyView from 'components/ItemsList/EmptyView';

const List = ({ data }) => {
  const results = data.map(({ title, thumbnail, href, ingredients }) => {
    return (
      <Item
        title={title}
        thumbnail={thumbnail}
        href={href}
        ingredients={ingredients}
        key={shortid.generate()}
      />
    );
  });

  return (
    <Container>
      {data.length ? results : <EmptyView />}
    </Container>
  );
};

List.propTypes = {
  data: PropTypes.array.isRequired
};

export default List;
