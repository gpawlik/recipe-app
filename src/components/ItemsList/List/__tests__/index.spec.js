import React from 'react';
import { create } from 'react-test-renderer';

import List from '../';

jest.mock('components/ItemsList/EmptyView', () => 'EmptyView');
jest.mock('components/ItemsList/Item', () => 'Item');

const initialProps = {
  data: []
};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <List {...finalProps} />
  );
};

describe('ItemsList/List', () => {
  it('should render an empty view with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render a list with custom props', () => {
    const props = {
      data: [{
        title: 'foo',
        thumbnail: 'bar',
        href: 'baz',
        ingredients: 'faz'
      }]
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
