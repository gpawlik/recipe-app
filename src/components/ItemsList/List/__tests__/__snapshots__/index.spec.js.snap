// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList/List should render a list with custom props 1`] = `
<ul
  className="sc-bdVaJa kHlqHg"
>
  <Item
    href="baz"
    ingredients="faz"
    thumbnail="bar"
    title="foo"
  />
</ul>
`;

exports[`ItemsList/List should render an empty view with the default props 1`] = `
<ul
  className="sc-bdVaJa kHlqHg"
>
  <EmptyView />
</ul>
`;
